### ISSUE TITLE
Should ActiveFedora implement LDP containment principles?

### ISSUE THREAD

**Issue by awead** [projecthydra/active_fedora#782]
What if you could do:
``` ruby
class SomeItems < ActiveFedora:Base
  has_direct_container Box
end
```
``` ruby
class Box < ActiveFedora:Base
  is_direct_container_for SomeItems, membership: :hasItems
end
```
And then have similar methods, `has_basic_container` and `has_indirect_container`.

**Comment by awead** [comment]
See also #714

**Comment by awead** [comment]
This would refactor/replace #780

**Comment by no-reply** [comment]
:+1: on something like this.

**Comment by terrellt** [comment]
:+1: This would be great, and was something I was looking to put in activefedora-aggregations, but I think you're right that it belongs here.

**Comment by cbeer** [comment]
:+1:  to doing it, but the example code does't look right to me. I'm also not sure the thing-that-contains-containers needs to know what type of container it has, does it?

**Comment by terrellt** [comment]
@cbeer Seems like that's the case where you'd want it to know, so that it could initialize the container on creation. 

I don't really understand what `is_direct_container_for` does, though.

**Comment by awead** [comment]
@terrellt @cbeer Yeah, that was just a mock-up, so my code example is probably wrong. I wondered if we needed `is_direct_container_for` as well and now thinking on it more it seems unnecessary.

**Comment by terrellt** [comment]
I think a good start would be

```ruby
class GenericFile < ActiveFedora::Base
   has_direct_container :files, :member_relation => RDF::DC.hasPart
end
```

Which will spin up a /files direct container on save with a MemberRelation of RDF::DC.hasPart and a membershipResource of GenericFile's #id, when it's created.

I think `has_` implies that the membershipResource is the containing resource, but maybe i'm wrong? It seems like if it's not then the statement would sit somewhere else.

**Comment by cbeer** [comment]
@terrellt :+1:

**Comment by awead** [comment]
@terrellt yep!

**Comment by awead** [comment]
This has been implemented in:
3facdba2b7a07cebde590bdbce976b376d0f8e76 (direct containers)
7a36f432ef2023b0fce0a876d29d72aefc596b44 (indirect containers)
